### PR TITLE
Refactoring peers db repository - Closes #1419

### LINKED ISSUE #1419: Refactor peers db repository
Parent: #1376 

This one is specifically for the peers repository.

### PR DESCRIPTION
**Pull request by vitaly-t**
Closes #1419

### CODE REVIEW

File: modules/peers.js [hunk 1]
```diff
@@ -293,17 +293,18 @@ __private.dbSave = function (cb) {
...
+		yield t.peers.clear();
+		yield t.peers.insert(peers);
+	})
+		.then(() => {
+			library.logger.info('Peers exported to database');
+			return setImmediate(cb);
+		})
+		.catch(err => {
+			library.logger.error('Export peers to database failed', {error: err.message || err});
+			return setImmediate(cb);
```
Comment (nazarhussain) [inline modules/peers.js]: @vitaly-t For now we are only using ES6 inside DB repositories. I believe we should avoid mixing ES6 code in modules for now. Yes we will migrate all code to ES6 all together at some stage.

Comment (nazarhussain) [reply modules/peers.js]: @karmacoma What your intake on it?

File: modules/peers.js [hunk 2]
```diff
@@ -225,7 +225,7 @@ __private.insertSeeds = function (cb) {
 		});
 	}, function (err) {
 		library.logger.trace('Peers->insertSeeds - Peers discovered', {updated: updated, total: library.config.peers.list.length});
-		return setImmediate(cb);
+		return setImmediate(cb, err);
```
Comment (nazarhussain) [inline modules/peers.js]: @4miners This code seems related to you. Was it intentionally that you missed sending error to top level chain?

File: modules/peers.js (hunk 1 above)
Comment (vitaly-t) [reply modules/peers.js]: Once we got green light on ES6 within branch 1.0.0, no point holding back. And besides, once I refactor a task or transaction, moving it into ES6 land is the only proper solution.

File: modules/peers.js (hunk 2 above)
Comment (4miners) [reply modules/peers.js]: Few points:
- Fail to update status of single peer not means that function `insertSeeds` failed entirely
- Even if error is propagated here - it's discarded at top level function (`async.series`)

So adding error propagation here not change behavior.

File: modules/peers.js [hunk 3]
```diff
@@ -267,11 +267,11 @@ __private.dbLoad = function (cb) {
 			}
 		}, function (err) {
 			library.logger.trace('Peers->dbLoad Peers discovered', {updated: updated, total: rows.length});
-			return setImmediate(cb);
+			return setImmediate(cb, err);
```
Comment (4miners) [inline modules/peers.js]: We will never get `err` here, because we not propagating it from inside `peer.rpc.status`.

File: modules/peers.js [hunk 4]
```diff
@@ -267,11 +267,11 @@ __private.dbLoad = function (cb) {
...
 		});
 	}).catch(function (err) {
 		library.logger.error('Import peers from database failed', {error: err.message || err});
-		return setImmediate(cb);
+		return setImmediate(cb, err);
```
Comment (4miners) [inline modules/peers.js]: That is not change of behavior because `err` is discarded at to level function (`async.series`).

Comment (vitaly-t) [reply modules/peers.js]: So I will leave as is ;)

File: modules/peers.js (hunk 3 above)
Comment (vitaly-t) [reply modules/peers.js]: Will leave it as is then ;)

**Comment by 4miners** [comment]
Regarding errors propagation, we call all those functions here:
https://github.com/LiskHQ/lisk/blob/3aa88fbcc62067e9dfc64f12e287ce1c60f1464d/modules/peers.js#L545-L564
As you see `async.series` there will not stop because of errors, as they are ignored. That is expected behavior. For example - inside test environment, or using local network all those functions should fail because we don't have any peers that we can connect with. But we still needs to fire `peersReady` event for loader module.

**Comment by vitaly-t** [comment]
@4miners I have reverted that change of mine ;) Thank you for the feedback! 👍

**Comment by 4miners** [comment]
@vitaly-t At least error from failed query should be propagated.

**Pull request merged**